### PR TITLE
HPCC4J-401 Provides proper JavaDoc support at proj/package level

### PR DESCRIPTION
**Pull request by rpastrana**
- Provides proper Header/footer
- Provides overview javadoc content at wsclient, commons, and dfsclient
- Provides package level content
- Excludes generated code documentation
- Updates wsclient,commons class level content

Signed-off-by: Rodrigo Pastrana <Rodrigo.Pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu please review

**Comment by rpastrana** [comment]
@g-pan as discussed offline, please provide grammar, stylistic review.

File: wsclient/src/main/java/org/hpccsystems/ws/client/BaseHPCCWsClient.java [hunk 1]
```diff
@@ -31,6 +31,11 @@
 import org.hpccsystems.ws.client.wrappers.EspSoapFaultWrapper;
 import org.w3c.dom.Document;
 
+/**
+ * Defines functionality common to all HPCC WS Clients.
+ * Typically implemented by specific HPCC Web service clients.
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/BaseHPCCWsClient.java:36]: HPCC Systems

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java [hunk 2]
```diff
@@ -169,6 +170,10 @@ public Stub getDefaultStub() throws AxisFault
     }
 
     // from HPCC-Platform/dali/dfu/dfuwu.hpp DFUfileformat
+    /**
+     * Used to declare variable data format of file to be sprayed
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java:173]: period after sprayed.

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java [hunk 3]
```diff
@@ -80,11 +80,10 @@
 import org.w3c.dom.NodeList;
 
 /**
- * Provides soap client for HPCC WsDFU web service.
+ * Facilitates Logical File System actions on target HPCC instance.
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java:83]: HPCC Systems instance.

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUXRefClient.java [hunk 4]
```diff
@@ -43,7 +59,10 @@
 import org.hpccsystems.ws.client.wrappers.gen.wsdfuxref.DFUXRefUnusedFilesResponseWrapper;
 
 /**
- * Client for HPCC's WsDFUXRef web service
+ * Facilitates XRef Logical File System actions on target HPCC instance.
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUXRefClient.java:62]: HPCC Systems instace.

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsResourcesClient.java [hunk 5]
```diff
@@ -35,7 +35,8 @@ HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
 import org.hpccsystems.ws.client.wrappers.gen.wsresources.ServiceQueryResponseWrapper;
 
 /**
- * Client for HPCC's HPCCWsResources web service
+ * Facilitates discovery of containerized HPCC Systems resources
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsResourcesClient.java:38]: add period after resources.

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsStoreClient.java [hunk 6]
```diff
@@ -40,8 +40,13 @@
 import org.hpccsystems.ws.client.wrappers.ArrayOfEspExceptionWrapper;
 
 /**
- * Use as soap client for HPCC WsFileIo web service.
- * This includes creating a new file, and appending data to a file in the given SHPCC System.
+ * Facilitates access to HPCC key/value based Storage.
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsStoreClient.java:43]: does it make sense to use HPCC Systems key/value based Storage. (and should storage be init lower case)

File: wsclient/src/main/java/org/hpccsystems/ws/client/antlr/package-info.java [hunk 7]
```diff
@@ -0,0 +1,5 @@
+/**
+ * Used internally to parse various formats of ECL record definitions
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/antlr/package-info.java:2]: Add period after definitions.

File: wsclient/src/main/java/org/hpccsystems/ws/client/extended/package-info.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Provides  HPCC Systems client libraries used to actuate non-standard HPCC Web Service actions
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/extended/package-info.java:2]: remove extra space between Provides and HPCC   and add period after actions.

File: wsclient/src/main/java/org/hpccsystems/ws/client/extended/package-info.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+ *
+ * Extension of the org.hpccsystems.ws.client package for non-standard web services
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/extended/package-info.java:4]: period after services.

File: wsclient/src/main/java/org/hpccsystems/ws/client/package-info.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
+/**
+ * <pre>
+ * Provides several HPCC Systems client libraries used to actuate a variety of actions on
+ * a target HPCC Systems instance.
+ *
+ * These clients abstract connectivity, soap, and proprietary web service details away from
+ * the caller. Allowing the caller to focus on higher level logic without the need to account
+ * for web service details and interface version changes.
+ *
+ * Each of the HPCC Web Service clients expose specialized HPCC-centric actions, and the caller
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/package-info.java:10]: HPCC Systems Web Service clients

File: wsclient/src/main/java/org/hpccsystems/ws/client/package-info.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+ * can choose to interact with individual clients directly, or can interact with the multi-purpose HPCCWsClient.
+ * The caller must provide connection information including target HPCC address, ECLWatch port (typically 8010 or 18010),
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/package-info.java:12]: HPCC Systems address,

File: wsclient/src/main/java/org/hpccsystems/ws/client/package-info.java [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+ * and user credentials.
+ *
+ * Create connection:
+ * org.hpccsystems.ws.client.utils.Connection connection = new Connection("http://myhpcc:8010");
+ *
+ * Provide HPCC user credentials:
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/package-info.java:18]: ? HPCC Systems user credentials:

File: wsclient/src/main/java/org/hpccsystems/ws/client/platform/package-info.java [hunk 13]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Provides local representation of target HPCC Systems platform
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/platform/package-info.java:2]: add period after platform.

File: wsclient/src/main/java/org/hpccsystems/ws/client/platform/test/package-info.java [hunk 14]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Used internally as base class for remote tests
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/platform/test/package-info.java:2]: ? should tests have a period. following? remote tests.

File: wsclient/src/main/java/org/hpccsystems/ws/client/utils/CollectionDelta.java [hunk 15]
```diff
@@ -11,6 +11,10 @@
 import java.util.Arrays;
 import java.util.Collection;
 
+/**
+ * Used internally to monitor and track changes in generic collections
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/utils/CollectionDelta.java:15]: add period after collections.

File: wsclient/src/main/java/org/hpccsystems/ws/client/utils/Connection.java [hunk 16]
```diff
@@ -16,6 +16,13 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+/**
+ * Represents and structures connection information
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/utils/Connection.java:20]: add period after information.

File: wsclient/src/main/java/org/hpccsystems/ws/client/utils/DatasetParser.java [hunk 17]
```diff
@@ -18,6 +18,11 @@
 import org.xml.sax.InputSource;
 import org.xml.sax.SAXException;
 
+/**
+ * Provides generic support for parsing of ECL Dataset responses from
+ * HPCC Engines.
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/utils/DatasetParser.java:23]: ? HPCC Systems Engines. (perhaps Engines should have init cap?)

File: wsclient/src/main/java/org/hpccsystems/ws/client/utils/FileFormat.java [hunk 18]
```diff
@@ -2,6 +2,10 @@
 
 import java.util.HashMap;
 
+/**
+ * Enumeration and convenience methods of known HPCC File formats
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/utils/FileFormat.java:6]: HPCC Systems file formats. (add period after formats - also check capitalization)

File: wsclient/src/main/java/org/hpccsystems/ws/client/utils/Utils.java [hunk 19]
```diff
@@ -37,6 +37,10 @@
 import org.w3c.dom.NodeList;
 import org.xml.sax.SAXException;
 
+/**
+ * Provides multiple functions which support HPCCWsClient actions
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/utils/Utils.java:41]: ??should this have a period after actions.

File: wsclient/src/main/java/org/hpccsystems/ws/client/utils/package-info.java [hunk 20]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Useful utilities for users of the HPCCWsClient libraries
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/utils/package-info.java:2]: Period after libraries.

File: wsclient/src/main/java/org/hpccsystems/ws/client/utils/package-info.java [hunk 21]
```diff
@@ -0,0 +1,8 @@
...
+ *
+ * Provided utilities include generic connection class, HPCC dataset
+ * response parser, HPCC file format enumerations, and more.
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/utils/package-info.java:5]: HPCC Systems file format

File: wsclient/src/main/java/org/hpccsystems/ws/client/wrappers/package-info.java [hunk 22]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Layer of abstraction between caller and HPCC Web Service interface
```
Comment (g-pan) [inline wsclient/src/main/java/org/hpccsystems/ws/client/wrappers/package-info.java:2]: HPCC Systems Web Service interface. (Period after interface)

**Review by g-pan — COMMENTED** [review]
See comments in line

**Comment by rpastrana** [comment]
@g-pan all comments have been addressed, please confirm.

**Review by g-pan — COMMENTED** [review]
One more comment.

**Comment by jpmcmu** [comment]
@rpastrana Looks good to me. @g-pan was super thorough

**Pull request merged**